Package constants (#8916)
FirebirdSQL/firebird: doc/sql.extensions/README.packages.txt

      tables that the package body depends on that object. If you want to, for example, drop that
      object, you first need to remove who depends on it. As who depends on it is a package body,
      you can just drop it even if some other database object depends on this package. When the body
      is dropped, the header remains, allowing you to create its body again after changing it based
      on the object removal.

      A package constant is a value initialized by a constant expression.
      A constant expression is defined by a simple rule: its value does not change after recompilation.
      Constants declared in the package specification are publicly visible and can be referenced using
      the <package>.<constant_name> notation.
      Constants declared in the package body are private and cannot be accessed from outside the package.

>       However, they can be referenced directly by <constant_name> within <procedure_impl> and <function_impl>.

Reviewer: This appears that package header constant cannot be referenced in its own body without the qualified package name.

Author: I corrected the desription

      tables that the package body depends on that object. If you want to, for example, drop that
      object, you first need to remove who depends on it. As who depends on it is a package body,
      you can just drop it even if some other database object depends on this package. When the body
      is dropped, the header remains, allowing you to create its body again after changing it based
      on the object removal.

      A package constant is a value initialized by a constant expression.
      A constant expression is defined by a simple rule: its value does not change after recompilation.
      Constants declared in the package specification are publicly visible and can be referenced using
      the [<schema>.]<package>.<constant_name> notation.
      Constants declared in the package body are private and cannot be accessed from outside the package.
      However, they can be referenced directly by <constant_name> within <procedure_impl> and <function_impl>.

>       Header constants can also be used directly with just the name for package body elements.

Reviewer 2: ```suggestion
      Header constants can also be referenced directly by their name inside package body elements.
```

        GRANT SELECT ON TABLE secret TO PACKAGE pk_secret;
        GRANT EXECUTE ON PACKAGE pk_secret TO ROLE role_secret;

>       To use package constants in a select expression, a USAGE permission is requeued:

Reviewer 2: ```suggestion
      To use package constants in a select expression, a USAGE permission is required:
```

>       To use package constants in a select expression, a USAGE permission is required:

Reviewer: I suppose not only in a "select expression", but in any place outside the package it was defined.

Author: Yes, the word "select" is redundant
